HW13/HomeWork13.cpp

```diff
@@ -0,0 +1,245 @@
+#include <iostream>
+#include <time.h>
+#include <string>
+#include <fstream>
+
+
+// Loading DataBase
+
+void LoadDataBase(const char* FileName, std::string* DataBase, int DataBaseSize)
+{
+	std::ifstream File;
+	File.open(FileName, std::ios::in);
+
+	if (!File)
+	{
+		std::cout << "File not found!" << std::endl;
+	}
+	else
+	{
+		//File >> std::noskipws;
+		for (int i = 0; i < DataBaseSize && !File.eof(); i++)
+		{
+			std::getline(File, DataBase[i]);
+		}
+	}
+	File.close();
+}
+
+// Wordle Algorithm and Wordle Init
+
+bool CheckMatch(const std::string& PlayerWord, std::string& Word)
+{
+	int MatchCount = 0;
+	std::string Result(5, '*');
+
+	for (int i = 0; i < Word.size(); i++)
+	{
+		Word[i] = std::toupper(Word[i]);
+		if (Word[i] == PlayerWord[i])
+		{
+			Result[i] = Word[i];
+			MatchCount++;
+		}
+		else
+		{
+			for (int j = 0; j < PlayerWord.size(); j++)
+			{
+				if (Word[i] == PlayerWord[j] && Result[j] == '*')
+				{
+					Result[j] = std::tolower(PlayerWord[j]);
+					break;
+				}
+			}
+		}
+	}
+	std::cout << "RESULT : " << Result << std::endl;
```

Comment:
The output shouldn't be part of the algorithm. You already modify the `result` string so can freely output it outside